packages/flutter_tools/lib/src/asset.dart

```diff
@@ -317,7 +319,9 @@ DevFSContent _createFontManifest(Map<String, dynamic> manifestDescriptor,
     fonts.addAll(manifestDescriptor['fonts']);
   if (fonts.isEmpty)
     return null;
-  return new DevFSStringContent(JSON.encode(fonts));
+  String json = JSON.encode(fonts);
+  json = json.replaceAll(_assetJsonLeadingParentRegExp, '"asset":"');
```

Comment:
Created another pr for option 2
#8943

Not sure how I feel about creating a half generic converter though. Feels like there are more assumptions in the converter than in the json string manipulator.

Created https://github.com/dart-lang/yaml/issues/26 which I would feel more comfortable in the long run.